src/commonMain/kotlin/io/github/detekt/sarif4k/SarifSerializer.kt

```diff
@@ -15,4 +20,54 @@ object SarifSerializer {
     fun toJson(sarif: SarifSchema210): String = json.encodeToString(sarif)
 
     fun fromJson(json: String): SarifSchema210 = Json.decodeFromString(json)
-}
\ No newline at end of file
+}
+
+// Extracted from https://github.com/Kotlin/kotlinx.serialization/issues/296#issuecomment-1859572541
+private fun Any?.toJsonElement(): JsonElement = when (this) {
```

Comment:
Considering this is very likely to be in an inner loop (serialisation), should the `when` be ordered by likelihood? e.g. null, String, map, list, etc, arrays.

Reply:
I moved the arrays down, I agree that they are not likely to be called that often (probably never).